src/main/java/jenkins/plugins/openstack/compute/slaveopts/BootSource.java

```diff
@@ -354,6 +354,21 @@ public String toString() {
                 return "Volume from Image";
             }
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+            if (o == null || getClass() != o.getClass())
+                return false;
+            final VolumeFromImage that = (VolumeFromImage) o;
+            return name.equals(that.name);
+        }
+
+        @Override
+        public int hashCode() {
+            return name.hashCode();
```

Comment:
This doesn't include all fields.
Use the parent class's hashCode as a basis and then incorporate this class's fields into the mix.

Reply:
Agreed these implementations are not correct. I am ok living without them or have it correct.